test/integration/client-side-encryption/client_side_encryption.prose.test.js

```diff
@@ -1723,6 +1723,148 @@ describe('Client Side Encryption Prose Tests', metadata, function () {
+      await client.close();
+    });
+
+    context('Case 1', metadata, function () {
+      it('createDataKey() handles duplicate key errors on the keyvault collection', async function () {
+        // 1. Use client_encryption to create a new local data key with a keyAltName "abc" and assert the operation does not fail.
+        try {
+          await clientEncryption.createDataKey('local', {
+            keyAltNames: ['abc']
+          });
+        } catch (err) {
+          expect.fail(
+            'Error in step 1) expected clientEncryption.createDataKey to not fail, but received',
+            err
+          );
+        }
+
+        // 2. Repeat Step 1 and assert the operation fails due to a duplicate key server error (error code 11000).
+        try {
+          await clientEncryption.createDataKey('local', {
+            keyAltNames: ['abc']
+          });
+          expect.fail(
+            'Error in step 2) expected clientEncryption.createDataKey to throw duplicate key error but it did not'
+          );
+        } catch (err) {
+          expect(err).have.property('code', 11000);
+        }
+
+        // 3. Use client_encryption to create a new local data key with a keyAltName "def" and assert the operation fails due to a duplicate key server error (error code 11000).
+        try {
+          await clientEncryption.createDataKey('local', {
+            keyAltNames: ['def']
+          });
+          expect.fail(
+            'Error in step 3) expected clientEncryption.createDataKey to throw duplicate key error but it did not'
+          );
+        } catch (err) {
+          expect(err).to.have.property('code', 11000);
+        }
+      });
+    });
+
+    context('Case 2', metadata, function () {
+      it('addKeyAltName() handles duplicate key errors on the keyvault collection', async function () {
+        let _id;
+        // 1. Use client_encryption to create a new local data key and assert the operation does not fail.
+        try {
+          _id = await clientEncryption.createDataKey('local');
+        } catch (err) {
+          expect.fail(
+            'Error in step 1) expected clientEncryption.createDataKey to not fail, but received',
+            err
+          );
+        }
+
+        // 2. Use client_encryption to add a keyAltName "abc" to the key created in Step 1 and assert the operation does not fail.
+        try {
+          await clientEncryption.addKeyAltName(_id, 'abc');
+        } catch (err) {
+          expect.fail(
+            'Error in step 2) expected clientEncryption.createDataKey not to fail, but received',
+            err
+          );
+        }
+
+        // 3. Repeat Step 2, assert the operation does not fail, and assert the returned key document contains the keyAltName "abc" added in Step 2.
+        try {
+          const { keyAltNames } = await clientEncryption.addKeyAltName(_id, 'abc');
+          expect(keyAltNames).to.contain('abc');
+        } catch (err) {
+          expect.fail(
```

Comment:
I think this transforms the potential error in L1836 in an undesirable way, i.e., instead of getting the expectation failure for keyAltNames, we're instead getting an error that says the method call failed

elsewhere we've adopted a `await method.catch(err => err)` pattern, which could be the solution here and elsewhere in the test code, but for an immediate fix that keeps the pattern you are using the same, you should postpone the destructuring that you do in L1835, and check the property immediately after the try/catch block